lib/auth/trustedcluster.go

```
@@ -67,8 +68,15 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, trustedCluster types.
 	}
 
 	// change state
+
+	if exists && !cmp.Equal(existingCluster.GetRoleMap(), trustedCluster.GetRoleMap()) {
+		if err := a.checkLocalRoles(ctx, trustedCluster.GetRoleMap()); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		a.updateUserCARoleMap(ctx, existingCluster, trustedCluster)
```

Returned error is ignored.